kyuubi-events/src/main/scala/org/apache/kyuubi/events/handler/KafkaLoggingEventHandler.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.events.handler
+
+import java.util.Properties
+
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.events.KyuubiEvent
+
+/**
+ * This event logger logs Kyuubi engine events to Kafka.
+ */
+class KafkaLoggingEventHandler(
+    topic: String,
+    producerConf: Properties,
+    kyuubiConf: KyuubiConf) extends EventHandler[KyuubiEvent]
+  with Logging {
+
+  private val kafkaProducer: KafkaProducer[String, String] = {
+    // set default serializer
+    if (!producerConf.containsKey("key.serializer")) {
+      producerConf.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
+    }
+    if (!producerConf.containsKey("value.serializer")) {
+      producerConf.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer")
+    }
+
+    new KafkaProducer[String, String](producerConf)
+  }
+
+  override def apply(event: KyuubiEvent): Unit = {
+    try {
+      val record = new ProducerRecord[String, String](topic, event.eventType, event.toJson)
```

Comment:
I remember that Kafka allows setting event timestamp on constructing `ProducerRecord`, it may be useful in some cases.

Reply:
Good suggestion. Setting timestamp with `System.currentTimeMillis`. Is that good enough? Or use Java8's `Instant.now.toEpochMilli` (with UTC clock)?

Reply:
While timestamp is not set , the producer will assign the timestamp using `System.currentTimeMillis()`.

Reply:
Usually, it should be event time, but I just notice that KyuubiEvent does not have event time ...

Reply:
Yes, no available unified event time in KyuubiEvent right now.

Reply:
let's leave a TODO here, we should add an `event_time` in `KyuubiEvent` and propagate it to Kafka, but it may break the backward compatibility, so better to do it later separately